### PR TITLE
fix(middleware): added lists route to be accessible by logged out users

### LINKED ISSUE #2457: The "Lists" tab on the hub seems broken for non-authenticated users
> The "Lists" tab on the hub seems broken in this for me: it redirects to the highlights home:

~~Are you logged in?~~

Just checked now, and it also appears broken for me... I don't think this is related to this PR, but i'll open a ticket for it

_Originally posted by @OgDev-01 in https://github.com/open-sauced/app/issues/2194#issuecomment-1885104309_
            

https://github.com/open-sauced/app/assets/62995161/07345425-4c50-4f35-a79b-46f3c455ef3e

### PR DESCRIPTION
**Pull request by sourabpramanik**
## Description
This PR fixes broken lists tab for non-authenticated user by adding lists route from being checked for user's session so that logged out users can access the demo list

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [X] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
fix #2457

## Mobile & Desktop Screenshots/Recordings
Before
bug

After
fix

## Steps to QA
1. Log out
2.  Go to lists page
3. You can view the demo list and no more getting redirected to feeds page

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [X] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

No

### CODE REVIEW

**Review by nickytonline — APPROVED** [review]
Thanks for the fix @sourabpramanik! 🚀

File: middleware.ts [hunk 1]
```diff
@@ -48,7 +48,8 @@ export async function middleware(req: NextRequest) {
   if (
     !pathsToMatch.some((matcher) => pathToRegexp(matcher).test(req.nextUrl.pathname)) ||
     // if the path is hub/insights go to the page so logged out users can see demo insights
```
[the remark this answers is not in the document]
Comment (sourabpramanik) [reply middleware.ts:50]: sure

**Review by nickytonline — APPROVED** [review]
:shipit:

**Review by OgDev-01 — APPROVED** [review]

File: middleware.ts [hunk 2]
```diff
@@ -47,8 +47,8 @@ export async function middleware(req: NextRequest) {
 
   if (
     !pathsToMatch.some((matcher) => pathToRegexp(matcher).test(req.nextUrl.pathname)) ||
-    // if the path is hub/insights go to the page so logged out users can see demo insights
-    req.nextUrl.pathname === "/hub/insights"
+    // if the path is hub/insights or hub/lists go to the page so logged out users can see demo insights or demo lists    req.nextUrl.pathname === "/hub/insights" ||
```
Comment (OgDev-01) [inline middleware.ts:50]: > This part (req.nextUrl.pathname === "/hub/insights" ||) is actually included in the comment

Thanks for catching this @ThePiyushAggarwal. looks kinda tricky to detect at first glance :<


@sourabpramanik please remove check condition from the commented code

**Review by OgDev-01 — CHANGES_REQUESTED** [review]

**Comment by sourabpramanik** [comment]
Thanks for catching that. @ThePiyushAggarwal

**Review by jpmcb — APPROVED** [review]
👍🏼 🕺🏼

**Review by OgDev-01 — APPROVED** [review]

**Pull request merged**